pgmpy refactor

ankurankan opened the pull request.

pandaabinash committed:
Renamed requirements file.

ankurankan committed:
fit method for BayesianModel [refs #224]

ankurankan committed:
minor change in docstring

ankurankan committed:
fixed pep8

ankurankan committed:
corrected fit method [ref #224]

ankurankan committed:
corrected fit method [refs #224]

ankurankan committed:
adds docstring [refs #224]

ankurankan committed:
corrected fit and other function [refs #224]

ankurankan committed:
corrected fit method [refs #224]

ankurankan committed:
a very dirty hack to make TabularCPD's __init__ to work with both python lists and numpy arrays

ankurankan committed:
fixed fit method [refs #224]

ankurankan committed:
docstring edit

ankurankan committed:
adds pandas to requirements and update numpy version

ankurankan committed:
removed breakpoint

ankurankan committed:
dirty hack

ankurankan committed:
tests for fit method [INCOMPLETE] [refs #224]

pandaabinash committed:
[ref-229] Inplace argument added in the Factor methods.

pandaabinash committed:
[ref-230] Moved JunctionTree class from base to models module.

pandaabinash committed:
Merge branch 'issue224' of https://github.com/ankurankan/pgmpy into dev

pandaabinash committed:
Renamed Inference module to inference module.

pandaabinash committed:
[refs-230] Refactored pgmgpy codebase:
    - to create base class for DirectedGraph and UndirectedGraph
    - to change base class for BayesianModel and MarkovModel

pandaabinash committed:
Renaming Inference module to inference module in doctests.

pandaabinash committed:
[refs-230] Removed unnecessary imports from __init__.py.

pandaabinash committed:
Added class for FactorGraph.

pandaabinash committed:
Created new JunctionTree class in pgmpy.models module.

pandaabinash committed:
Added imports of JunctionTree and FactorGraph in __init__.py of models module.

pandaabinash committed:
Renaming of Inference module into inference module in tests_inference.py

pandaabinash committed:
Removed commented-out tests from test_factor.py

pandaabinash committed:
Merge branch 'dev' of https://github.com/abinashpanda/pgmpy into dev

pandaabinash committed:
Modified BayesianModel.py to-
    * fix bug in get_indpendencies
    * fix bug in conversion to markov model

pandaabinash committed:
Created methods for finding local independencies and markov blanket.

pandaabinash committed:
Docstring changes in Independencies class.

pandaabinash committed:
Added method for finding variable nodes and factor nodes in FactorGraph class.

pandaabinash committed:
Added method for computing partition function in UndirectedGraph class.

pandaabinash committed:
Created docstring in get_independencies() method for BayesianModel.

pandaabinash committed:
Modified get_local_independencies() method in MarkovModel to include latex
input argument.

pandaabinash committed:
Minor refactoring in the __init__ method of TabularCPD.

pandaabinash committed:
Moved the docstring of class from __init__ method to class definition in
Inference module.

pandaabinash committed:
Modified get_factors() method in:
    - JunctionTree
    - FactorGraph

pandaabinash committed:
Added method for conversion of markov model to bayesian model.

pandaabinash committed:
Bug fixes in __init__ method in CPD class.

pandaabinash committed:
Initial commit of tests for DirectedGraph class.

pandaabinash committed:
Modified:
    - DirectedGraph to have moralize method
    - BayesianModel to use moralize method to convert to MarkovModel

pandaabinash committed:
Initial commit of tests for DirectedGraph along with changes in the docstrings.

pandaabinash committed:
Modified MarkovModel to fix bug in
    - get_local_independencies()
    - to_junction_tree()
methods.

pandaabinash committed:
Initial commit of tests of MarkovModel class.

pandaabinash committed:
Modified tests for BayesianModel according to the refactoring.

pandaabinash committed:
Corrected tests for Inference model according to refactoring.

pandaabinash committed:
Skipped tests for factor_divide

pandaabinash committed:
Modified .travis.yml to install pandas

ankurankan commented on pgmpy/base/DirectedGraph.py:
```diff
@@ -1,238 +1,290 @@
+    def __init__(self, ebunch=None):
+        super(DirectedGraph, self).__init__(ebunch)
+        self.cpds = []
+        self.cardinalities = defaultdict(int)
```
Why are you associating cpds and cardinalities with the base DirectedGraph class? Shouldn't it be a properties of model classes? I think DirectedGraph should just have the basic structure details of the network.

And things like structures for storing the states of the variables etc should be in the DirectedGraph

abinashpanda replied:
Instead of having `self.cpds` in `DirectedGraph`, we can have something like `self.factors` as we don't know whether the factors associated with the _directed graphical model_ would be a CPD or not. Furthermore as the nodes in the `DirectedGraph` class represents some random variable, factors or a cluster of factors, we can associate cpds and cardinalities to `DirectedGraph` class. Just a suggestion.

pandaabinash committed:
Docstring fixes in triangulate() method in UndirectedGraph class.

abinashpanda commented:
- Both `UndirectedGraph` and `DirectedGraph` serves as an intermediate class between pure `networkx` graph classes and `pgmpy` models. All the methods that can be applied to all the _undirected graphical models_ should reside in `UndirectedGraph` and all the methods that can be applied to all the _directed graphical models_ should reside in `DirectedGraph` class.
- For example, `triangulate()` method should reside in `UndirectedGraph` class as any _undirected graphical model_ can be triangulated.

pandaabinash committed:
Merge branch 'dev' of https://github.com/pgmpy/pgmpy into dev

pandaabinash committed:
Added API docs for base class as well the newly created model and fixed doc
string issues.

pandaabinash committed:
- Removed sphinxext submodule
- Created a directory for all the sphinx extensions we would be using.

pandaabinash committed:
Modified conf.py to fix the issues created by new sphinxext.

pandaabinash committed:
Initial commit for framework for adding examples.

pandaabinash committed:
Proper separtion of each chapter in pgmpy documentation page.

pandaabinash committed:
Modified requirements.txt and conf.py to use sphinx_bootstrap_theme.

pandaabinash committed:
[ref-231] Removed factors and cardinalities with all corresponding methods from
UndirectedGraph model.

pandaabinash committed:
[ref-231] Refactored MarkovModel class to have all the methods for factor
operations and cardinalities of the variables.

pandaabinash committed:
- Initial commit for tests of FactorGraph class and methods
- Refactoring in FactorGraph to have all the factor related operations over
  here.

pandaabinash committed:
- Initial commit for tests of JunctionTree classes and methods
- Refactoring in JunctionTree to have all the factor operation over here.

pandaabinash committed:
Refactored DirecteGraph along with tests to remove CPDs from it.

pandaabinash committed:
Refactored BayesianModel along with tests to have CPDs over here.

pandaabinash committed:
Modified requirements.txt to use gen_rst.
